### DISCUSSION TITLE
Can't retrieve variable stored in AWS SecretsManager in DAG in MWAA (Airflow 2.5.1)

### DISCUSSION THREAD

**Discussion by joao-miranda** [apache/airflow#31535] [Q&A]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

Airflow 2.5.1

I've followed the instructions I found online, for example https://blog.beachgeek.co.uk/working-with-parameters-and-variables-in-amazon-managed-workflows-for-apache-airflow/
I'm storing a variable encrypted by KMS in SecretsManager with name **airflow/variables/AAA.**

 
This configuration is present in the Airflow Environment:
```
secrets.backend | airflow.providers.amazon.aws.secrets.secrets_manager.SecretsManagerBackend
secrets.backend_kwargs | { "connections_prefix": "airflow/connections", "variables_prefix": "airflow/variables" }
```

Airflow IAM user has full KMS and SecretsManager permissions.

Trying to get the value from the code like this is not working:
```
AAA = Variable.get('AAA')
```

The DAG import fails with this error:
```
Broken DAG: [/usr/local/airflow/dags/pipeline.py] Traceback (most recent call last):
  File "/usr/local/airflow/dags/pipeline.py", line 14, in <module>
    AAA = Variable.get('AAA')
  File "/usr/local/airflow/.local/lib/python3.10/site-packages/airflow/models/variable.py", line 141, in get
    raise KeyError(f"Variable {key} does not exist")
KeyError: 'Variable AAA does not exist'
```

### What you think should happen instead

Value should be retrieved from Secrets Manager

### How to reproduce

The above code in MWAA should replicate the same behaviour.

### Operating System

AWS

### Versions of Apache Airflow Providers

AWS MWAA (Apache Airflow 2.5.1)

### Deployment

Amazon (AWS) MWAA

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by eladkal** [comment]
The report doesn't clarify what is the issue.
What exactly is the bug?

**Comment by joao-miranda** [comment]
The bug is that integration between Airflow and AWS SecretsManager seems to be broken.

**Comment by eladkal** [comment]
Please add reproduce example

**Comment by vincbeck** [comment]
Are you sure the AWS connection points to the right account and region? The message `Variable AAA does not exist` makes me thing the AWS connection targets and account/region where the variable is not defined

**Comment by manugarri** [comment]
@vincbeck i am experiencing the same issue during a migration from 2.4.3. 

I am testing 2.7.2 with the same secrets airflow.cfg

```[secrets]
# Full class name of secrets backend to enable (will precede env vars and metastore in search path)
# Example: backend = airflow.providers.amazon.aws.secrets.systems_manager.SystemsManagerParameterStoreBackend
backend=airflow.providers.amazon.aws.secrets.secrets_manager.SecretsManagerBackend


# The backend_kwargs param is loaded into a dictionary and passed to __init__ of secrets backend class.
# See documentation for the secrets backend you are using. JSON is expected.
# Example for AWS Systems Manager ParameterStore:
# ``{{"connections_prefix": "/airflow/connections", "profile_name": "default"}}``
backend_kwargs={"connections_prefix" : "", "variables_prefix" : "", "region_name": "us-east-1", "config_prefix":null}

```

This configuration allow our dags to read variables from aws secretmanager on 2.4.3 , however on 2.7.2 the dag fails to access the variable and throws the exception ``` File "/usr/local/airflow/.local/lib/python3.11/site-packages/airflow/models/variable.py", line 140, in get
    raise KeyError(f"Variable {key} does not exist")```

Have the backend_kwargs changed in recent versions?

**Reply by Taragolis** [reply]
Change your [log severity level](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#logging-level) to DEBUG and have a look what is your output, and maybe this one might help you to find actual error

**Reply by manugarri** [reply]
**UPDATE** I realised the issue was the docker airflow image for 2.7.2 was not reading the local `airflow.cfg`  I rebuilt the image from scratch and the secrets backend works now.
